### PR TITLE
Cleaned Up Middleware Stack Code

### PR DESCRIPTION
**Pull request by sampocs**
Closes: #XXX

## Context and purpose of the change
Now that the middleware stacks have grown since inception, their definition in `app.go` has become slightly confusing for someone not familiar with middleware. This PR does a simple cleanup to try and rename/ organize in a way that's slightly easier to follow.

**Question for reviewer**: I'm not super familiar with the purpose of the IBC routing. Why do we not add a route for `recordstypes.ModuleName -> recordsStack`? I assumed that's because packets never flow directly to records, but if that was the case, I wouldn't expect a route for stakeibc either.

## Brief Changelog
* Variable renaming / re-ordering / commenting to follow 3 stacks: ICAHost, ICAController, and Records

## Author's Checklist

I have...

- [x] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: app/app.go
Comment (asalzmann) [inline app/app.go:562]:
```suggestion
	// Define the IBC middleware stacks
```

File: app/app.go
Comment (asalzmann) [inline app/app.go:571]: ICAController, icacallbacks and stakeibc are all middlewares

File: app/app.go
Comment (asalzmann) [inline app/app.go:578]: We should probably call this the transfer stack, for clarity

File: app/app.go
Comment (asalzmann) [inline app/app.go:576]: nit: is there a nice way to format this as a single line of code? something like
```
icamiddlewarestack := icacontroller.NewIBCModule(app.ICAControllerKeeper,
                                    icacallbacksmodule.NewIBCModule(app.IcacallbacksKeeper,
                                    stakeibcIBCModule)
```
this orders the stack in a visually intuitive way (not sure if the formatting works)

File: app/app.go
Comment (asalzmann) [inline app/app.go:588]: We should double check that changing the route order here won't cause unexpected changes in how packets are routed

File: app/app.go
Comment (asalzmann) [inline app/app.go:590]: nit: leave this note in / update with a clarifying comment (I think it answers the question for the reviewer you had)

**Review by asalzmann — CHANGES_REQUESTED** [review]
Couple of minor comments. This code is a bit finicky and not well-tested - should we add tests as part of this PR?

> Question for reviewer: I'm not super familiar with the purpose of the IBC routing. Why do we not add a route for recordstypes.ModuleName -> recordsStack? I assumed that's because packets never flow directly to records, but if that was the case, I wouldn't expect a route for stakeibc either.

The reason is that the middleware stacks for ICA modules in ibc-go v3 are designed in a confusing way. Typically, IBC middlewares sit on _top_ of other modules. For example, when we route incoming transfer packets, they are passed to the `recordsStack`, which contains the records module, which calls the transfer module.

However, ICAController authentication modules (like `stakeibc`) wrap ICAController. So, incoming ICA packets are routed to the stakeibc module, then to the ICAController module. This was flagged as confusing and addressed in ibc-go v5 (it's backwards compatible afaiu, so nothing in our current design should break).

File: app/app.go
Comment (sampocs) [reply app/app.go:589]: talked offline, ordering shouldn't matter since its just adding items to a map

File: app/app.go
Comment (sampocs) [reply app/app.go:590]: I just moved it up to the line above `ibcRouter := ibcporttypes.NewRouter()`

File: app/app.go
Comment (sampocs) [reply app/app.go:579]: my thinking was we should label each stack from either the top or bottom but be consistent about it. Which would mean we either do ICAControllerStack and RecordsStack, or StakeibcStack and TransferStack.

I thought using the top of the stack made the code slightly cleaner since you define them bottom up. What do you think?

File: app/app.go
Comment (sampocs) [reply app/app.go:577]: I'm not sure there's a way to do that cleanly. Ideally, we'd flop the order of arguments if the NewIBCModule functions so that each one shows `NewIBCModule(sameStack, newKeeper)` but we can't change the ICA ones.

The main change here I was trying to get at was changing the naming of `icamiddlewarestack` since that doesn't differentiate between ICAHost or ICAController.

Depending on how we want to name the stack (see earlier comment about ICAController vs Stakeibc), we can do something similar to how you had it before with
```go
var icaControllerStack ibcporttypes.IBCModule = stakeibcIBCModule
icaControllerStack = icacallbacksmodule.NewIBCModule(app.IcacallbacksKeeper, icaControllerStack)
icaControllerStack = icacontroller.NewIBCModule(app.ICAControllerKeeper, icaControllerStack)
```

**Comment by sampocs** [comment]
@asalzmann, I'll wait until we have the app-router and ratelimit merged in, but I think we should go with the following format:

```go
var stakeibcStack ibcporttypes.IBCModule = stakeibcIBCModule
stakeibcStack = icacallbacksmodule.NewIBCModule(app.IcacallbacksKeeper, stakeibcStack)
stakeibcStack = icacontroller.NewIBCModule(app.ICAControllerKeeper, stakeibcStack)


var transferStack ibcporttypes.IBCModule = transferIBCModule
transferStack = recordsmodule.NewIBCModule(app.RecordsKeeper, transferStack)
transferStack = ratelimitmodule.NewIBCModule(app.RateLimitKeeper, transferStack)
transferStack = routermodule.NewIBCModule(app.AppRouterKeeper, transferStack)
```

**Comment by sampocs** [comment]
Open

**Comment by sampocs** [comment]
closing in favor of #618

**Pull request closed without merging**
